### PR TITLE
DOC-161 Added plugins directory, pages, and added to sidebar for 24.2

### PR DESCRIPTION
**Pull request by rmalik-es**
Created plugins directory with individual markdown pages for our plugins:

- LDAP 
- Logs Endpoint
- OTLP 
- User certificates

**Other additions/changes:**
- All plugin pages added and copyedited with some changes in formatting. 
- A placeholder README file has been added. 
- Plugin section on the left nav bar of server docs has also been added.

This needs to be added to the 24.2 branch.

### CODE REVIEW

File: docs/plugins/LDAP-plugin.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+### 'Server certificate error' or 'Connect Error - The authentication or decryption has failed': 
+- Verify that the server certificate is valid. If it is a self-signed certificate, set `ValidateServerCertificate` to `false`.
+
+### The LDAP server is unavailable:
+
+-   Verify connectivity to the LDAP server from an EventStoreDB node (e.g. using `netcat` or `telnet`)
+-   Verify the `Host` and `Port` parameters
+-   Verify that the server certificate is valid. If it is a self-signed certificate, set `ValidateServerCertificate` to `false`.
+
+### Error authenticating with LDAPS server. System.AggregateException:
```
Comment (cdevarenne) [inline docs/plugins/LDAP-plugin.md:62]:
```suggestion
### Error authenticating with LDAP server. System.AggregateException: 
```

**Review by cdevarenne — CHANGES_REQUESTED** [review]

**Review by cdevarenne — APPROVED** [review]

File: docs/plugins/LDAP-plugin.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+### Exception during search - 'No such Object' or 'The object does not exist': 
+- Verify the `BaseDn`.
+
+### 'Server certificate error' or 'Connect Error - The authentication or decryption has failed': 
+- Verify that the server certificate is valid. If it is a self-signed certificate, set `ValidateServerCertificate` to `false`.
+
+### The LDAP server is unavailable:
+
+-   Verify connectivity to the LDAP server from an EventStoreDB node (e.g. using `netcat` or `telnet`)
```
Comment (rsowald) [inline docs/plugins/LDAP-plugin.md:58]:
```suggestion
-   Verify connectivity to the LDAP server from an EventStoreDB node (e.g. using `netcat` or `telnet`).
```

File: docs/plugins/LDAP-plugin.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+-   Verify the `Host` and `Port` parameters
```
Comment (rsowald) [inline docs/plugins/LDAP-plugin.md:59]:
```suggestion
-   Verify the `Host` and `Port` parameters.
```

File: docs/plugins/LDAP-plugin.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+-   Verify that the server certificate is valid. If it is a self-signed certificate, set `ValidateServerCertificate` to `false`.
+
+### Error authenticating with LDAP server. System.AggregateException: 
+
+- Example error message: `One or more errors occurred. ---> System.NullReferenceException: Object reference not set to an instance of an object. at Novell.Directory.Ldap.Connection.connect(String host, Int32 port, Int32 semaphoreId)`
+-   Due to a packaging bug, this error might occur when setting `UseSSL: true` on Windows. Extract `Mono.Security.dll` to the _EventStore_ folder (where _EventStore.ClusterNode.exe_ is located) as a workaround.
```
Comment (rsowald) [inline docs/plugins/LDAP-plugin.md:65]:
```suggestion
-   This packaging error might occur when setting `UseSSL: true` on Windows. Extract `Mono.Security.dll` to the _EventStore_ folder (where _EventStore.ClusterNode.exe_ is located) as a workaround. 
```

File: docs/plugins/LDAP-plugin.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+-   Verify that the server certificate is valid. If it is a self-signed certificate, set `ValidateServerCertificate` to `false`.
+
+### Error authenticating with LDAP server. System.AggregateException: 
+
+- Example error message: `One or more errors occurred. ---> System.NullReferenceException: Object reference not set to an instance of an object. at Novell.Directory.Ldap.Connection.connect(String host, Int32 port, Int32 semaphoreId)`
+-   Due to a packaging bug, this error might occur when setting `UseSSL: true` on Windows. Extract `Mono.Security.dll` to the _EventStore_ folder (where _EventStore.ClusterNode.exe_ is located) as a workaround. 
+
+### No errors in server logs but cannot login:
+
+-   Verify the `ObjectClass` and `Filter` parameters
```
Comment (rsowald) [inline docs/plugins/LDAP-plugin.md:69]:
```suggestion
-   Verify the `ObjectClass` and `Filter` parameters.
```

File: docs/plugins/LDAP-plugin.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+-   If you have set `RequireGroupMembership` to `true`, verify that the user is part of the group specified by `RequiredGroupDn` and that the LDAP record has the `memberOf` attribute (specified by `GroupMembershipAttribute`).
```
Comment (rsowald) [inline docs/plugins/LDAP-plugin.md:70]:
```suggestion
-   If you have set `RequireGroupMembership` to `true`, verify that the user is part of the group specified by `RequiredGroupDn` and that the LDAP record has `GroupMembershipAttribute` set to `memberOf`.
```

File: docs/plugins/Logs-endpoint-plugin.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
+# Logs Endpoint Plugin 
+The Logs Endpoint plugin provides HTPP access to EventStoreDB log files, facilitating easy log management.
```
Comment (rsowald) [inline docs/plugins/Logs-endpoint-plugin.md:2]:
```suggestion
The Logs Endpoint plugin provides HTTP access to EventStoreDB log files, facilitating easy log management. 
```

File: docs/plugins/Logs-endpoint-plugin.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
+# Logs Endpoint Plugin 
+The Logs Endpoint plugin provides HTPP access to EventStoreDB log files, facilitating easy log management. 
+
+::: tip
+The Logs Endpoint plugin is included in commercial versions of EventStoreDB.
+:::
+
+## Enabling the Plugin
+
+1. Download and Install: First, download the plugin and place it in the `plugins` folder within your EventStoreDB installation directory.
```
Comment (rsowald) [inline docs/plugins/Logs-endpoint-plugin.md:10]:
```suggestion
1. Download and install: First, download the plugin and place it in the `plugins` folder within your EventStoreDB installation directory. 
```

Comment (rsowald) [reply docs/plugins/Logs-endpoint-plugin.md:10]: The other docs use sentence case in the quick start steps.

File: docs/plugins/Logs-endpoint-plugin.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+```bash
+curl https://<USER>:<PASSWORD>@hostname-or-ip:2113/admin/logs | jq
+```
+Example:
+```bash
+
+curl https://user:password@localhost:2113/admin/logs | jq
+```
+
+This should show something similar to:
```
Comment (rsowald) [inline docs/plugins/Logs-endpoint-plugin.md:49]:
```suggestion
Sample response:
```

File: docs/plugins/Logs-endpoint-plugin.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+  },
+  {
+    "name": "log20240205.json",
+    "lastModified": "2024-02-05T13:14:37.0781601+00:00",
+    "size": 158542
+  }
+]
+```
+
+The response is ordered as most recent change first and limited to a maximum of 1000 items, and include:
```
Comment (rsowald) [inline docs/plugins/Logs-endpoint-plugin.md:65]:
```suggestion
The response is ordered from most recent change first, is limited to a maximum of 1000 items, and includes:
```

File: docs/plugins/Logs-endpoint-plugin.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+| Name | Description |
+|---|---|
+| name | File name of the log file |
+| createdAt | Timestamp of when the log file was created in [round-trip format](https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings#the-round-trip-o-o-format-specifier) (local time with time zone information) |
+| lastModified | Timestamp of the last modification in [round-trip format](https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings#the-round-trip-o-o-format-specifier) (local time with time zone information) |
+| size | Size of the log file in bytes |
+
+### Downloading Log Files
+
+To download a specific log file, append its name to the URL:
```
Comment (rsowald) [inline docs/plugins/Logs-endpoint-plugin.md:76]: @timothycoleman is the --output flag required or does it use the file name as the default?

File: docs/plugins/Logs-endpoint-plugin.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+To verify the plugin setup:
+
+1. Go to the EventStoreDB installation directory where the executable is located. 
+2. Ensure there's a directory called `plugins`. If not, create one. 
+3. Inside `plugins`, create a subfolder for the Logs Endpoint Plugin (e.g., `EventStore.Diagnostics.LogsEndpointPlugin` or any name you prefer), if it's missing.
+4. Place the plugin binaries in this subfolder.
+
+## Troubleshooting
+
+- **Plugin Not Loaded:** Ensure the plugin is correctly placed in the subdirectory within the `plugins` directory of your server's installation folder. If the `plugins` directory or the specific plugin subdirectory doesn't exist, you need to create them and    place the plugin binaries there.
```
Comment (rsowald) [inline docs/plugins/Logs-endpoint-plugin.md:96]:
```suggestion
- **Plugin Not Loaded:** Ensure the plugin is correctly placed in the subdirectory within the `plugins` directory of your server's installation folder. If the `plugins` directory or the specific plugin subdirectory doesn't exist, you need to create them and place the plugin binaries there. 
```

File: docs/plugins/OTLP-plugin.md [hunk 13]
```diff
@@ -0,0 +1,80 @@
+﻿# OTLP Exporter Plugin 
+
+The OTLP Exporter Plugin allows you to export EventStoreDB metrics to a specified endpoint using the OpenTelemetry Protocol (OTLP). This guide will help you set up the exporter and customize its configuration to meet your monitoring needs, so you can receive, process, and export metrics as desired. Learn more about the [OTEL collector here.](https://opentelemetry.io/docs/collector/)
```
Comment (rsowald) [inline docs/plugins/OTLP-plugin.md:3]:
```suggestion
The OTLP Exporter Plugin allows you to export EventStoreDB metrics to a specified endpoint using the OpenTelemetry Protocol (OTLP). This guide will help you set up the exporter and customize its configuration so you can receive, process, and export and monitor metrics as desired. Learn more about the [OTEL collector here.](https://opentelemetry.io/docs/collector/)
```

File: docs/plugins/OTLP-plugin.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+```
+2. Configuration file setup: You must also provide a JSON configuration file. This file should be placed in the `config` folder inside your EventStoreDB installation directory. If the `config` folder doesn't exist, create it and add your configuration file there. 
+
+
+## Configuration File Details 
+The configuration file should specify: 
+
+| Name     | Description                                                 |
+|----------|-------------------------------------------------------------|
+| Endpoint | Destination where the OTLP exporter will send the data.     |
```
Comment (rsowald) [inline docs/plugins/OTLP-plugin.md:25]: Other files do not have periods on sentence fragment descriptions.

File: docs/plugins/OTLP-plugin.md [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+| Headers  | Optional headers for the connection.                        |
+
+Headers are key-value pairs separated by commas. To provide custom headers, you can add the `Headers` option in the JSON configuration file and set the value according to your requirements. For example:
+```
+"Headers": "api-key=value,other-config-value=value"
+```
+In the above example, we are adding two custom key value pairs to the `Headers` option.
```
Comment (rsowald) [inline docs/plugins/OTLP-plugin.md:32]: This feels redundant and out of place. Can you incorporate it into the sentence above the example?

File: docs/plugins/OTLP-plugin.md [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+
+## Example Configuration File
```
Comment (rsowald) [inline docs/plugins/OTLP-plugin.md:34]: I don't think this needs to be a separate header. I would just remove this line.

File: docs/plugins/User-certificates-plugin.md [hunk 17]
```diff
@@ -0,0 +1,225 @@
+# User Certificates Plugin 
+The User Certificates plugin enhances security by allowing authentication through both a username/password and an x509 user certificate. User certificates work across any cluster with a shared root Certificate Authority (CA) with the user's certificate. This means that you can have a single user certificate that is valid across multiple clusters.
+
+::: tip
+The User Certificates plugin is included in commercial versions of EventStoreDB.
+:::
+
+
+# Setup Instructions
```
Comment (rsowald) [inline docs/plugins/User-certificates-plugin.md:9]:
```suggestion
## Setup Instructions
```

File: docs/plugins/User-certificates-plugin.md [hunk 18]
```diff
@@ -0,0 +1,225 @@
...
+
+1. Download the User Certificates plugin.
+2. Place it in the `.\plugins` directory within your EventStoreDB installation.
```
Comment (rsowald) [inline docs/plugins/User-certificates-plugin.md:12]:
```suggestion
2. Place it in the `.\plugins` directory within your EventStoreDB installation directory. 
```

File: docs/plugins/User-certificates-plugin.md [hunk 19]
```diff
@@ -0,0 +1,225 @@
...
+::: tip
+The User Certificates plugin is included in commercial versions of EventStoreDB.
+:::
+
+
+# Setup Instructions
+
+1. Download the User Certificates plugin.
+2. Place it in the `.\plugins` directory within your EventStoreDB installation. 
+3. In the EventStoreDB `.\config` directory, add the following configuration to `.\config\plugin-config.json`:
```
Comment (rsowald) [inline docs/plugins/User-certificates-plugin.md:13]:
```suggestion
3. Find the file `.\config\plugin-config.json` in the EventStoreDB `.\config` directory, add the following configuration:
```

File: docs/plugins/User-certificates-plugin.md [hunk 20]
```diff
@@ -0,0 +1,225 @@
...
+The user certificate must adhere to the following requirements:
+
+- The certificate has a root CA in common with the node certificate.
+- The root CA that they have in common is trusted by the node.
+- The CN of the certificate is the same as the username.
+- The certificate has the ClientAuth EKU, and not the ServerAuth EKU.
+- The certificate must be in date.
+- The CN is the username of a user that exists in the database.
+
+> Note: Certificates can also authenticate `admin` and `ops` users.
```
Comment (rsowald) [inline docs/plugins/User-certificates-plugin.md:52]: this `>` will just make a blockquote. I suggest turning this into a `::: tip` box

File: docs/plugins/User-certificates-plugin.md [hunk 21]
```diff
@@ -0,0 +1,225 @@
...
+Cert Hash(sha256): a37767096f10697fb0fcf94a3872d0984aeed4754a295c5320679876e00def32
+Signature Hash: 6d922badaba2372070f13c69b620286262eab1d8d2d2156a271a1d73aaaf64e4
+```
+
+</details>
+
+
+## Restrictions
+
+There is currently a limitation with this plugin that requests authenticated with user certificates cannot be forwarded between nodes.
```
Comment (rsowald) [inline docs/plugins/User-certificates-plugin.md:177]:
```suggestion
There is currently a limitation with this plugin: requests authenticated with user certificates cannot be forwarded between nodes.
```

File: docs/plugins/User-certificates-plugin.md [hunk 22]
```diff
@@ -0,0 +1,225 @@
...
+The User Certificates plugin cannot be used alongside any of the other authentication plugins, such as the LDAPS plugin or the OAuth plugin.
+
+## Troubleshooting
+
+### Plugin Not Loaded
+Ensure the plugin is correctly placed in the subdirectory within the `plugins` directory of your server's installation folder. If the `plugins` directory or the specific plugin subdirectory doesn't exist, you need to create them and place the plugin binaries there. To verify the plugin setup:
+
+1. Go to the EventStoreDB installation directory where the executable is located. 
+2. Ensure there's a directory called `plugins`. If not, create one. 
+3. Inside `plugins`, create a subfolder for the OTLP Exporter Plugin (e.g.,`EventStore.Auth.UserCertificates` or any name you prefer), if it's missing.
```
Comment (rsowald) [inline docs/plugins/User-certificates-plugin.md:190]:
```suggestion
3. Inside `plugins`, create a subfolder for the User Certificates Plugin (e.g.,`EventStore.Auth.UserCertificates` or any name you prefer), if it's missing.
```

File: docs/plugins/User-certificates-plugin.md [hunk 23]
```diff
@@ -0,0 +1,225 @@
...
+
+```
+[12824, 1,17:30:04.921,INF] Plugins path: "C:\eventstoredb-24.2.0-rc.1-ce-windows.x64\plugins"
+[12824, 1,17:30:04.922,INF] Adding: "C:\eventstoredb-24.2.0-rc.1-ce-windows.x64\plugins" to the plugin catalog.
+[12824, 1,17:30:04.924,INF] Adding: "C:\eventstoredb-24.2.0-rc.1-ce-windows.x64\plugins\EventStore.Auth.UserCertificates" to the plugin catalog.
+[12824, 1,17:30:05.253,INF] Loaded SubsystemsPlugin plugin: "user-certificates" "24.2.0.1".
+```
+
+### Plugin Not Started
+The plugin has to be configured to start to authenticate user requests.
```
Comment (rsowald) [inline docs/plugins/User-certificates-plugin.md:203]:
```suggestion
The plugin has to be configured to start authenticating user requests.
```

**Comment by rsowald** [comment]
Other than the suggestions in line, I have one nitpick you can do with what you will: the casing on headers is not consistent. For example, some H3s use sentence case and others use title case. I would choose one, even it is just applies to H3s and you want to keep all title casing for H1s and H2s.

**Review by timothycoleman — CHANGES_REQUESTED** [review]
i think the subdirectory thing isn't working correctly with the nav - if i navigate to one of the plugins pages then the links in the nav bar back to the other pages don't work.

its probably easiest to keep all the md in the same directory and prefix the filename with plugins or something like that

```
plugins-ldap.md
plugins-otlp.md
```
(sidenode: probably no need for any capitals)

File: docs/plugins/Logs-endpoint-plugin.md (hunk 8 above)
Comment (rmalik-es) [reply docs/plugins/Logs-endpoint-plugin.md:10]: Good eye

File: docs/plugins/User-certificates-plugin.md (hunk 20 above)
Comment (rmalik-es) [reply docs/plugins/User-certificates-plugin.md:52]: Good eye @rsowald

**Review by cdevarenne — APPROVED** [review]

File: docs/plugins-logs-endpoint.md [hunk 24]
```diff
@@ -0,0 +1,103 @@
...
+| Name | Description |
+|---|---|
+| name | File name of the log file |
+| createdAt | Timestamp of when the log file was created in [round-trip format](https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings#the-round-trip-o-o-format-specifier) (local time with time zone information) |
+| lastModified | Timestamp of the last modification in [round-trip format](https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings#the-round-trip-o-o-format-specifier) (local time with time zone information) |
+| size | Size of the log file in bytes |
+
+### Downloading Log Files
+
+To download a specific log file, append its name to the URL:
```
Comment (rsowald) [inline docs/plugins-logs-endpoint.md:76]:
```suggestion
To download a specific log file, append its name to the URL and name the downloaded file with an output flag:
```

**Pull request closed without merging**